### ISSUE TITLE
Syntactic sugar for retries.

### ISSUE THREAD

**Issue by drob** [softprops/retry#11]
Thanks for a great library!

Would you be open to a PR that allowed policies to take `Future[T]`s by-name? This would be in addition to the `() => Future[T]` option offered now.

This would let clients write:
```scala
retry.Directly(3) { attemptSomething() }
```
instead of
```scala
retry.Directly(3) { () => {
  attemptSomething()
}}
```

I believe I can do this with a small addition to the `Policy` trait and with no breaking changes.

**Comment by drob** [comment]
Went ahead and played with it, turns out a bit of a hack is required to preserve backcompat given the way type erasure works. My attempt here: #12.

**Comment by drob** [comment]
Fixed with #12.
